librz/include/rz_il/rzil_vm.h

```diff
@@ -100,6 +100,54 @@ RZ_API RzILEffect *rz_il_evaluate_effect(RzILVM *vm, RzILOp *op, RzILOpArgType *
 
+// Handler for core theory opcode
+void *rz_il_handler_ite(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_var(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_unk(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+
+void *rz_il_handler_bitv(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_msb(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_lsb(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_ule(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_sle(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_neg(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_not(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_add(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_sub(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_mul(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_div(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_sdiv(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_mod(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_smod(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_shiftl(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_shiftr(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_logical_and(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_logical_or(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_logical_xor(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+
+void *rz_il_handler_b0(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_b1(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_and_(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_or_(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_inv(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_cast(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_append(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_concat(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+
+
+void *rz_il_handler_perform(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_set(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_jmp(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_goto(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_seq(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_branch(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+
+void *rz_il_handler_load(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+void *rz_il_handler_store(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
+
+//TODO: remove me when all the handlers are implemented
+void *rz_il_handler_unimplemented(RzILVM *vm, RzILOp *op, RzILOpArgType *type);
```

Comment:
i have removed them for a reason. just add the concat/append at the beginning of vm_layer.c

Reply:
The question is how to provide RzList as argument inside the opcode, bcs there is no argtype for it

Reply:
i think you need to edit the RzILOpConcat structure to have an RzList, but i would wait for @XVilka for his input. i'm not sure if that is the right way to implement this.